tests/integration/functions/runtime_context/custom_type_parameters_test.ts

```diff
@@ -1,38 +1,32 @@
-import { assert, assertEquals, assertExists, IsAny, IsExact } from "../../../src/dev_deps.ts";
-import { CanBe, CanBeUndefined } from "../../../src/test_utils.ts";
-import { assertEqualsTypedValues } from "../../../src/test_utils.ts";
-import { SlackFunctionTester } from "../../../src/functions/tester/mod.ts";
-import { DefineFunction, DefineType } from "../../../src/mod.ts";
-import { EnrichedSlackFunctionHandler } from "../../../src/functions/types.ts";
-import { Schema } from "../../../src/mod.ts";
+import {
+  assert, assertExists, IsAny, IsExact
+} from "../../../../src/dev_deps.ts";
+import { CanBe, CanBeUndefined, CannotBeUndefined, } from "../../../../src/test_utils.ts";
+import { DefineFunction, DefineType, Schema } from "../../../../src/mod.ts";
+import { DefineProperty } from "../../../../src/parameters/define_property.ts";
+import {
+  EnrichedSlackFunctionHandler,
+} from "../../../../src/functions/types.ts";
+import { SlackFunctionTester } from "../../../../src/functions/tester/mod.ts";
+import { assertEqualsTypedValues } from "../../../../src/test_utils.ts";
+import { InternalSlackTypes } from "../../../../src/schema/slack/types/custom/mod.ts";
 
 /**
- * Custom Function handler tests, exercising inputs of unwrapped typed object parameters
- * Unwrapped typed object parameters are object parameters that are not defined using the DefineProperty helper
- * TODO: some of these are commented out and failing as we attempt to address the issues
- * TODO: May want to split this file up down the road if it gets too big, possibly organize under further sub-dirs
+ * Custom Function handler tests, exercising Custom Type inputs/outputs, including built-in/internal Slack custom types
  */
-
-
-// TODO: unwrapped typed object, in a custom type, fed to a typed array, yields an array of `any` items at runtime
-/*
-Deno.test("Custom Function using Typed Arrays of Custom Types of unwrapped typed objects should honor required and optional properties", () => {
-  const obj = {
+Deno.test("Custom Function using a Custom Type input for a DefineProperty-wrapped typedobject with mixed required/optional properties should provide correct typedobject typing in a function handler context and should complain if typedobject output does not include required property", () => {
+  const myObject = DefineProperty({
```

Comment:
Should we be using `DefineProperty` for Custom Type examples?

Reply:
No, we shouldn't, but I wanted to make sure it works!

Should I remove them?

Reply:
Yes please!

Reply:
Done in 169620e

Reply:
Oh actually for this one, I think we should keep the test but instead of using `DefineProperty`, we should just define the Custom Type directly

```ts
  const myType = DefineType({
    name: "custom",
    type: Schema.types.object,
    properties: {
      required_property: { type: "string" },
      optional_property: { type: "string" },
    },
    required: ["required_property"],
  });
```

Reply:
Yeah I think the test on line 137 in this file has that covered, but if not happy to add another one.